Examples/Python/src/Svg.cpp

```diff
@@ -199,6 +202,66 @@ std::vector<actsvg::svg::object> drawDetector(
   return svgDetViews;
 }
 
+struct SurfaceArrayCollector {
+  std::vector<std::tuple<Acts::GeometryIdentifier, const Acts::SurfaceArray*>>
+      surfaceArrays;
+  // Visitor pattern to collect the surface arrays
+  void operator()(const Acts::TrackingVolume* tVolume) {
+    // This is trying Gen1 first
+    const auto& cLayers = tVolume->confinedLayers();
+    if (cLayers != nullptr) {
+      for (const auto& layer : cLayers->arrayObjects()) {
+        if (layer != nullptr) {
+          const auto& surfaceArray = layer->surfaceArray();
+          if (surfaceArray != nullptr) {
+            surfaceArrays.emplace_back(layer->geometryId(), surfaceArray);
+          }
+        }
+      }
+    }
+    // Now trey Gen2
+    auto sArrayPolicy = dynamic_cast<const Acts::SurfaceArrayNavigationPolicy*>(
+        tVolume->navigationPolicy());
+    if (sArrayPolicy != nullptr) {
+      surfaceArrays.emplace_back(tVolume->geometryId(),
+                                 &sArrayPolicy->surfaceArray());
+    }
```

Comment:
There's one complication here: the policy attached to the volume itself is most likely going to be a `MultiNavigationPolicy`, that internally holds a try-all policy + the surface array one.

I have changes in-flight for the detray conversion that should make it possible to traverse this to get into the actual surface array policy.

I think it's fine to keep it like this for now(won't crash because it's guarded by the `if`, but maybe add a comment that this will need some follow up work.